pkg/ddevapp/config_test.go

```diff
@@ -1317,3 +1319,220 @@ func TestDatabaseConfigUpgrade(t *testing.T) {
 		assert.Empty(app.MariaDBVersion)
 	}
 }
+
+// TestConfigMergeStringList verifies that string merges update w/o clobber
+func TestConfigMergeStringList(t *testing.T) {
+	assert := asrt.New(t)
+	pwd, _ := os.Getwd()
+
+	projDir, err := filepath.Abs(testcommon.CreateTmpDir(t.Name()))
+	require.NoError(t, err)
+
+	err = fileutil.CopyDir("./testdata/TestConfigMergeItems/.ddev", filepath.Join(projDir, ".ddev"))
+	require.NoError(t, err)
+
+	app, err := NewApp(projDir, true)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		err = os.Chdir(pwd)
+		assert.NoError(err)
+		err = app.Stop(true, false)
+		assert.NoError(err)
+		err = os.RemoveAll(projDir)
+		assert.NoError(err)
+	})
+	err = os.Chdir(app.AppRoot)
+	assert.NoError(err)
+
+	// test matches allowing for delete syntax (a prefixed !)
+	assertSimpleMatch := func(expected bool, match string, setting []string) {
+		test := assert.True
+		if !expected {
+			test = assert.False
+		}
+
+		for _, val := range setting {
+			if val == match {
+				test(true, match)
+				return
+			}
+		}
+		test(false, match)
+	}
+	// no clobber of old setting
+	assertSimpleMatch(true, "somename", app.AdditionalHostnames)
+	// successful merge
+	assertSimpleMatch(true, "somename-new", app.AdditionalHostnames)
+	// do not want the delete instruction added to the list
+	assertSimpleMatch(false, "!someothername", app.AdditionalHostnames)
+
+}
+
+// TestConfigMergeEnvItems verifies that config overrides for web_environment
+// override and do not clobber settings from config.yaml.
+func TestConfigMergeEnvItems(t *testing.T) {
+	assert := asrt.New(t)
+	pwd, _ := os.Getwd()
+
+	projDir, err := filepath.Abs(testcommon.CreateTmpDir(t.Name()))
+	require.NoError(t, err)
+
+	err = fileutil.CopyDir("./testdata/TestConfigMergeItems/.ddev", filepath.Join(projDir, ".ddev"))
```

Comment:
Again, use t.Name() for your testdata access, thanks.

Reply:
But I imagine you don't have to do the copying at all, since NewApp() takes a project directory, so you can just NewApp(path/to/testdata). That would not be true if the test were messing with the app, but it's just reading it here.

Reply:
What I'd really like to do is refactor the set up here so that I'm not just copying the first half of the test.  Hell: I copied similar code from another test :-) But using t.Name() will resolve the issue to both our satisfaction, I think.  I'll do that.